### PR TITLE
healthcare plan schemas 

### PR DESCRIPTION
**Pull request by prabhum2**
- Created new healthcare plan class
- Created new healthcare plan details filed group extending healthcare plan class

### CODE REVIEW

File: components/classes/healthcare-plan.schema.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{
+  "meta:license": [
+    "Copyright 2022 Adobe Systems Incorporated. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/classes/healthcare-plan",
```
Comment (mjasrotia) [inline components/classes/healthcare-plan.schema.json:8]: Let's call this just "plan" as it's a general concept and be used for other industries too.

File: components/classes/healthcare-plan.schema.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          "Title": "Plan ID",
+          "type": "string",
+          "description": "Unique identifier of the insurance plan."
+        },
+        "xdm:planName": {
+          "Title": "Plan Name",
+          "type": "string",
+          "description": "Name of the insurance plan."
+        },
+        "xdm:planType": {
```
Comment (mjasrotia) [inline components/classes/healthcare-plan.schema.json:32]: If its only for primary, secondary kind of healthcare plans, let's move this to the healthcare-plan-details FG. We can also leave it here with no industry specific enums and add "isPrimary", isSecondary" and "isTeritiary" properties in the Field group for healthcare plan details.

File: components/fieldgroups/healthcare-plan/healthcare-plan-details.schema.json [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  "description": "Details of a Health Insurance plan provided by an organization.",
+  "definitions": {
+    "plan-details": {
+      "properties": {
+        "xdm:payerID": {
+          "Title": "Payer ID",
+          "type": "string",
+          "description": "Unique identifier of the payer."
+        },
+        "xdm:contractID": {
```
Comment (mjasrotia) [inline components/fieldgroups/healthcare-plan/healthcare-plan-details.schema.json:27]: Contract looks more like an agreement between Plan and a member so, it should be either removed from here or moved to the plans array in member if required.

File: components/fieldgroups/healthcare-plan/healthcare-plan-details.schema.json [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+          "format": "date-time",
+          "description": "Coverage end date."
+        },
+        "xdm:lastVerification": {
+          "Title": "Last Verification Date",
+          "type": "string",
+          "format": "date-time",
+          "description": "The date on which this plan was last verified."
+        },
+        "xdm:cost": {
```
Comment (mjasrotia) [inline components/fieldgroups/healthcare-plan/healthcare-plan-details.schema.json:77]: cost is a relative term which changes across insurance companies and the insured. Not sure if we need to add cost in plan. The same plan could be sold for a different cost depending upon the insured pool. Please check with SMEs if we need to move it somewhere else.

File: components/fieldgroups/healthcare-plan/healthcare-plan-details.schema.json [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+          "enum": ["Medical", "Dental", "Vision", "Accident"],
+          "meta:enum": {
+            "Medical": "Medical",
+            "Dental": "Dental",
+            "Vision": "Vision",
+            "Accident": "Accident"
+          },
+          "description": "Member plan coverage type."
+        },
+        "xdm:coverageStartDate": {
```
Comment (mjasrotia) [inline components/fieldgroups/healthcare-plan/healthcare-plan-details.schema.json:59]: Coverage Start and End Dates are Member-Plan specific These dates are already in the Member==>plans[] array so, we can remove from here.

File: components/fieldgroups/healthcare-plan/healthcare-plan-details.schema.json (hunk 4 above)
Comment (prabhum2) [reply components/fieldgroups/healthcare-plan/healthcare-plan-details.schema.json:60]: This is added based on SME's feedback

**Comment by mjasrotia** [comment]
Moved this to
https://github.com/adobe/xdm/pull/1563

**Pull request closed without merging**